test/semantics/dosemantics02.f90

```diff
@@ -28,3 +31,13 @@ IMPURE FUNCTION random() RESULT(i)
       i = 35
     END FUNCTION random
 END SUBROUTINE do_concurrent_c1121
+
+SUBROUTINE s1()
+  ! Warn since no constrains are violated 
+  DO 20 I = 1, 10, 0
+  20 CONTINUE
+
```

Comment:
Maybe one more test with a constant parameter variable.

Reply:
Great suggestion!  I'll do that.